Sources/NIOHTTP1Server/main.swift

```diff
@@ -33,6 +33,29 @@ extension String {
     }
 }
 
+private func httpResponseHead(request: HTTPRequestHead, status: HTTPResponseStatus, headers: HTTPHeaders = HTTPHeaders()) -> HTTPResponseHead {
+    var head = HTTPResponseHead(version: request.version, status: status, headers: headers)
+    if request.isKeepAlive && request.version.major == 1 && request.version.minor == 0 {
+        /* to work around `ab -k` behaving weirdly: it'll send an HTTP/1.0 request with keep-alive set
+           but stops doing anything at all if the server doesn't also set Connection: keep-alive */
+        switch head.headers["connection"] {
+        case let headers where headers.count > 1:
+            // bad, more than one connection header, let's sanitise
```

Comment:
There's nothing bad about multiple connection headers. You want `getCanonicalForm` (which I realise should have been spelled `[canonicalForm:]` but too late now.

Reply:
To be clear, we aren't saying you should only accept keep-alive, close here. I'm saying you should tolerate almost anything else here, you just only want to introspect for values `keep-alive` and `close`.

Reply:
@Lukasa as a side note should be mark `getCanonicalForm` as deprecated and add `[canonicalForm:]` ?